### DISCUSSION TITLE
Vitest throwing on the `document.queryCommandSupported` in the a dependency

### DISCUSSION THREAD

**Discussion by Lemour-sudo** [TypeFox/monaco-languageclient#614] [Q&A]
I am running vitest on my project and it throws an error "TypeError: document.queryCommandSupported is not a function" from the monaco-vscode-api library in: "node_modules/vscode/vscode/src/vs/editor/contrib/clipboard/browser/clipboard.js:16:92".
When opening the clipboard.js, I see that the `document.queryCommandSupported` has been deprecated: https://developer.mozilla.org/en-US/docs/Web/API/Document/queryCommandSupported

From the monaco-languageclient repo, I see there are some unit tests and running successfully.

Any suggestions as to how circumvent the vitest error?

**Comment by CGNonofr** [comment]
Is is possible to polyfill it in vitest? I think it's your best chance

**Comment by Lemour-sudo** [comment]
Does this not also mean that the monaco-vscode-api is using a deprecated method `document.queryCommandSupported` that should be replaced though? @CGNonofr

**Reply by CGNonofr** [reply]
[VSCode does](https://github.com/microsoft/vscode/blob/fcb8417ba39fabb5daced28aa54f316f228bff68/src/vs/editor/contrib/clipboard/browser/clipboard.ts#L27-L28). If it's an issue for you, I think that's where the issue should be reported

**Comment by Lemour-sudo** [comment]
Thanks! I was not aware that vs code was doing it. Thank you @CGNonofr
